docs/src/architecture/state.md

```diff
@@ -11,89 +11,60 @@ the system maintains 3 databases to describe the state:
-
-As with accounts, there would be a strong incentive to use private notes as they would result in lower fees. This is also beneficial to the network as a private note adds only 64 bytes to the state (32 bytes when it is produced, and 32 bytes when it is consumed).
+Using a Merkle Mountain Range (append-only accumulator) means that we can't remove individual elements from it. This seemingly means that the size of the note database would grow indefinitely. Moreover, at high tps, it would grow very quickly: at 1K tps we'd be adding about 1TB/year to the database.
 
-Using an append-only accumulator means that we can't remove individual elements from it. This would seemingly mean that the size of the note database would grow indefinitely. Moreover, at high tps, it would grow very quickly: at 1K tps we'd be adding about 1TB/year to the database.
+However, we need to explicitly store only the unconsumed public notes and enough info to construct membership proofs against them. Private notes, as well as public notes which have already been consumed, can be safely discarded. Such notes would still remain in the accumulator, but there is no need to store them explicitly as the append-only accumulator can be updated without knowing all items stored in it. This reduces actual storage requirements to a fraction of the database's nominal size.
 
-However, we need to explicitly store only the unconsumed public notes and enough info to construct membership proofs against them. Private notes, as well as public notes which have already been consumed, can be safely discarded. Such notes would still remain in the accumulator, but there is no need to store them explicitly as the append-only accumulator can be updated without knowing all items stored in it. This would reduce actual storage requirements to a fraction of the database's nominal size.
-
-Moreover, since notes are not meant as long-lived objects, we can impose some lifetimes restrictions on them. For example, we could say that the system will store only $2^{35}$ most recent notes (a note not consumed in this timeframe becomes un-spendable). This can be easily done by discarding old commitment roots once the number of roots exceeds 35. At 1K tps this would mean that notes would be discarded after about 1 year, but this would also mean that the size of the note database will never grow beyond about 1TB.
+> (Is this the case in v1?) Moreover, since notes are not meant as long-lived objects, we can impose some lifetimes restrictions on them. For example, we could say that the system will store only $2^{35}$ most recent notes (a note not consumed in this timeframe becomes un-spendable). This can be easily done by discarding old commitment roots once the number of roots exceeds 35. At 1K tps this would mean that notes would be discarded after about 1 year, but this would also mean that the size of the note database will never grow beyond about 1TB.
```

Comment:
how did you calculate the $2^35$ most recent notes will be preserved.

Reply:
Assuming a transaction on average consumes 1 note, we'd get about $2^{35}$ nullifiers generated per year. To compute storage requirements for these we need to make the following assumptions:

- How much storage is needed per nullifier.
- How long is a nullifier epoch.

If we assume each nullifier requires 32 bytes to store, we'd need about 1TB to store all $2^{35}$ nullifiers. Since nodes need to store the last two nullifiers epochs, if we make an epoch size 6 months, then we'd need 1TB of storage. So, one set of assumptions to get to this number is:

- 1K TPS.
- 1 nullifier per TX.
- 33 bytes per nullifier.
- 6 months per epoch.

But I'm actually not sure if we need 33 bytes to store nullifiers. I think this is true if we assume nullifiers are stored in a key-value map where keys are 32 bytes and values are 1 byte. But given that nullifier values are theoretically just 1 bit, there is probably a much more efficient way to store them.